src/main/java/spoon/reflect/declaration/CtFormalTypeDeclarer.java

```diff
@@ -43,4 +45,9 @@
 	 * Removes a type parameters from this generic element.
 	 */
 	boolean removeFormalCtTypeParameter(CtTypeParameter formalTypeParameter);
+
+	/**
+	 * @return the instance of {@link GenericTypeAdapter} which can adapt type parameters from different scopes to scope of this declarer
+	 */
+	GenericTypeAdapter getGenericTypeAdapter();
```

Comment:
> What about adding a spoon factory method instead? Should I use an AbstractSpoonVisitor instead of ugly if?

Excellent. What about `TypeFactory#createTypeAdapter()`?